// flippa/dist/Client.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = (function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var _superagent = require("superagent");

var _superagent2 = _interopRequireDefault(_superagent);

var _qs = require("qs");

var _qs2 = _interopRequireDefault(_qs);

var Client = (function () {
  function Client() {
    var opts = arguments.length <= 0 || arguments[0] === undefined ? {} : arguments[0];

    _classCallCheck(this, Client);

    this.baseEndpointURL = opts.baseEndpointURL || "https://api.flippa.com/v3";
    this.accessToken = opts.accessToken || null;
    this.timeout = opts.timeout || 15000;
  }

  _createClass(Client, [{
    key: "get",
    value: function get(endpoint) {
      var params = arguments.length <= 1 || arguments[1] === undefined ? {} : arguments[1];

      var request = _superagent2["default"].get(this.baseEndpointURL + endpoint).withCredentials().query(_qs2["default"].stringify(params)).set("Accept", "application/json").timeout(this.timeout);

      this._setAuthorizationHeader(request);

      return request;
    }
  }, {
    key: "post",
    value: function post(endpoint) {
      var params = arguments.length <= 1 || arguments[1] === undefined ? {} : arguments[1];
      var cookies = arguments.length <= 2 || arguments[2] === undefined ? {} : arguments[2];

      var request = _superagent2["default"].post(this.baseEndpointURL + endpoint).withCredentials().set("Accept", "application/json").set("Content-Type", "application/json").send(params).timeout(this.timeout);

      this._setCookieHeader(request, cookies);

      this._setAuthorizationHeader(request);

      return request;
    }
  }, {
    key: "del",
    value: function del(endpoint) {
      var params = arguments.length <= 1 || arguments[1] === undefined ? {} : arguments[1];

      var request = _superagent2["default"].del(this.baseEndpointURL + endpoint).withCredentials().set("Accept", "application/json").set("Content-Type", "application/json").send(params).timeout(this.timeout);

      this._setAuthorizationHeader(request);

      return request;
    }
  }, {
    key: "_setAuthorizationHeader",
    value: function _setAuthorizationHeader(request) {
      if (this.accessToken) {
        request.set("Authorization", "Bearer " + this.accessToken);
      }
    }
  }, {
    key: "_setCookieHeader",
    value: function _setCookieHeader(request, cookies) {
      if (Object.keys(cookies).length !== 0) {
        request.set("Cookie", this._cookieString(cookies));
      }
    }
  }, {
    key: "_cookieString",
    value: function _cookieString(cookies) {
      return Object.keys(cookies).reduce(function (acc, key) {
        return "" + acc + key + "=" + cookies[key] + ";";
      }, "");
    }
  }]);

  return Client;
})();

exports["default"] = Client;
module.exports = exports["default"];
