// flippa/dist/Flippa.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = (function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var _Client = require("./Client");

var _Client2 = _interopRequireDefault(_Client);

var _resourcesCareer = require("./resources/Career");

var _resourcesCareer2 = _interopRequireDefault(_resourcesCareer);

var _resourcesCareers = require("./resources/Careers");

var _resourcesCareers2 = _interopRequireDefault(_resourcesCareers);

var _resourcesLead = require("./resources/Lead");

var _resourcesLead2 = _interopRequireDefault(_resourcesLead);

var _resourcesListing = require("./resources/Listing");

var _resourcesListing2 = _interopRequireDefault(_resourcesListing);

var _resourcesListings = require("./resources/Listings");

var _resourcesListings2 = _interopRequireDefault(_resourcesListings);

var _resourcesMetrics = require("./resources/Metrics");

var _resourcesMetrics2 = _interopRequireDefault(_resourcesMetrics);

var _resourcesSalesPage = require("./resources/SalesPage");

var _resourcesSalesPage2 = _interopRequireDefault(_resourcesSalesPage);

var _resourcesSavedSearch = require("./resources/SavedSearch");

var _resourcesSavedSearch2 = _interopRequireDefault(_resourcesSavedSearch);

var _resourcesSavedSearchSubscription = require("./resources/SavedSearchSubscription");

var _resourcesSavedSearchSubscription2 = _interopRequireDefault(_resourcesSavedSearchSubscription);

var _resourcesSessionFlashMessages = require("./resources/SessionFlashMessages");

var _resourcesSessionFlashMessages2 = _interopRequireDefault(_resourcesSessionFlashMessages);

var _resourcesSessions = require("./resources/Sessions");

var _resourcesSessions2 = _interopRequireDefault(_resourcesSessions);

var _resourcesSupportEnquiries = require("./resources/SupportEnquiries");

var _resourcesSupportEnquiries2 = _interopRequireDefault(_resourcesSupportEnquiries);

var _resourcesUser = require("./resources/User");

var _resourcesUser2 = _interopRequireDefault(_resourcesUser);

var _resourcesUsers = require("./resources/Users");

var _resourcesUsers2 = _interopRequireDefault(_resourcesUsers);

var _resourcesPartnerPage = require("./resources/PartnerPage");

var _resourcesPartnerPage2 = _interopRequireDefault(_resourcesPartnerPage);

var _resourcesIntercom = require("./resources/Intercom");

var _resourcesIntercom2 = _interopRequireDefault(_resourcesIntercom);

var Flippa = (function () {
  function Flippa() {
    var opts = arguments.length <= 0 || arguments[0] === undefined ? {} : arguments[0];

    _classCallCheck(this, Flippa);

    this.client = new _Client2["default"](opts);
  }

  _createClass(Flippa, [{
    key: "authenticate",
    value: function authenticate(params) {
      var _this = this;

      var cookies = arguments.length <= 1 || arguments[1] === undefined ? {} : arguments[1];

      return new Promise(function (resolve, reject) {
        _this.client.post("/oauth2/token", params, cookies).then(function (res) {
          _this.client.accessToken = res.body.access_token;
          resolve(_this);
        })["catch"](function (err) {
          reject(err);
        });
      });
    }
  }, {
    key: "listing",
    value: function listing(listingId) {
      return new _resourcesListing2["default"](this.client, listingId);
    }
  }, {
    key: "user",
    value: function user(userId) {
      return new _resourcesUser2["default"](this.client, userId);
    }
  }, {
    key: "salesPage",
    value: function salesPage(salesPageId) {
      return new _resourcesSalesPage2["default"](this.client, salesPageId);
    }
  }, {
    key: "savedSearch",
    value: function savedSearch(savedSearchId) {
      return new _resourcesSavedSearch2["default"](this.client, savedSearchId);
    }
  }, {
    key: "savedSearches",
    value: function savedSearches() {
      return new _resourcesSavedSearch2["default"](this.client, null);
    }
  }, {
    key: "savedSearchSubscription",
    value: function savedSearchSubscription(savedSearchId) {
      return new _resourcesSavedSearchSubscription2["default"](this.client, savedSearchId);
    }
  }, {
    key: "savedSearchSubscriptions",
    value: function savedSearchSubscriptions() {
      return new _resourcesSavedSearchSubscription2["default"](this.client, null);
    }
  }, {
    key: "career",
    value: function career(careerId) {
      return new _resourcesCareer2["default"](this.client, careerId);
    }
  }, {
    key: "partnerPage",
    value: function partnerPage(pageName) {
      return new _resourcesPartnerPage2["default"](this.client, pageName);
    }
  }, {
    key: "accessToken",
    get: function get() {
      return this.client.accessToken;
    },
    set: function set(token) {
      this.client.accessToken = token;
    }
  }, {
    key: "listings",
    get: function get() {
      return new _resourcesListings2["default"](this.client);
    }
  }, {
    key: "metrics",
    get: function get() {
      return new _resourcesMetrics2["default"](this.client);
    }
  }, {
    key: "users",
    get: function get() {
      return new _resourcesUsers2["default"](this.client);
    }
  }, {
    key: "sessionFlashMessages",
    get: function get() {
      return new _resourcesSessionFlashMessages2["default"](this.client);
    }
  }, {
    key: "sessions",
    get: function get() {
      return new _resourcesSessions2["default"](this.client);
    }
  }, {
    key: "supportEnquiries",
    get: function get() {
      return new _resourcesSupportEnquiries2["default"](this.client);
    }
  }, {
    key: "careers",
    get: function get() {
      return new _resourcesCareers2["default"](this.client);
    }
  }, {
    key: "lead",
    get: function get() {
      return new _resourcesLead2["default"](this.client);
    }
  }, {
    key: "intercom",
    get: function get() {
      return new _resourcesIntercom2["default"](this.client);
    }
  }]);

  return Flippa;
})();

exports["default"] = Flippa;
;
module.exports = exports["default"];
